### DISCUSSION TITLE
Provide dynamic content for aside navigation items

### DISCUSSION THREAD

**Discussion by tsironis13** [taiga-family/taiga-ui#9101] [Q&A] [answered]
Hello,
Is there a way to provide a dynamic content using content projection for aside navigation items inside a `tuiNavigationAside` component?

<img width="317" alt="image" src="https://github.com/user-attachments/assets/974bb214-1524-42d3-9d05-c6984cdc2d37">

<img width="296" alt="image" src="https://github.com/user-attachments/assets/999f5d9d-a330-4c07-9cde-10a8c4439316">

I have tried to inject 2 aside items but the `tuiAsideItem` directives are not applied.

**Comment by waterplea** [comment, marked as answer]
Problem is `tuiAsideItem` needs to inject `tuiNavigationAside` to be able to watch for it expansion status and it cannot do that when `tuiNavigationAside` is in a wrapper, this is how DI in Angular works. You can try making your wrapper extend `tuiNavigationAside` and provide it like that:
https://stackblitz.com/edit/angular-kdxny3
Note that `tuiNavigationAside` also provides config for buttons and dropdowns which would also need to be copy-pasted, so this is a bit shaky since those might change — it's an internal implementation.
